### PR TITLE
Create basic Gin server

### PR DESCRIPTION
**Pull request by pdeziel**
This adds a basic gin server with initial configuration and a status endpoint.

### CODE REVIEW

**Review by pdeziel — COMMENTED** [review]
Most of this is ported over from our other projects - not too much unique happening yet.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+      - name: Lint Go Code
+        run: staticcheck ./...
+
+  test:
+    name: Go Test
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
```
Comment (pdamodaran) [inline .github/workflows/test.yml:35]: that's great you added the option to use multiple versions.

File: pkg/api/v1/client.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+var _ CourierClient = &APIv1{}
+
+//===========================================================================
+// Client Methods
+//===========================================================================
+
+// Status returns the status of the courier service.
+func (c *APIv1) Status(ctx context.Context) (out *StatusReply, err error) {
```
Comment (pdamodaran) [inline pkg/api/v1/client.go:56]: curious why the variable is named `c` instead of `s`

**Review by pdamodaran — APPROVED** [review]
nice work @pdeziel!  I just had one question but apart from that I think it's good to go!

File: pkg/api/v1/client.go (hunk 2 above)
Comment (pdeziel) [reply pkg/api/v1/client.go:56]: I was thinking "c" for client.

**Pull request merged**